R/util.R

```diff
@@ -83,6 +83,41 @@ util.duplicated_all <- function(x) {
   return (duplicated_downward | duplicated_upward)
 }
 
+util.na_omit <- function(x){
+  # returns all values of vector, matrix or data.frame x that are not NA,
+  # but maintains the type of x (in contrast with stats::na.omit which adds a
+  # bunch of unsolicited attributes to vectors)
```

Comment:
nit: I would add that when acting on 2D data, it throws out whole columns if those columns have any NA values; came as a slight surprise to me.

Reply:
yeah I didn't actually know about that but it would surprise me too